spec/compiler/type_inference/global_spec.cr

```diff
@@ -185,6 +185,20 @@ describe "Global inference" do
       )) { types["Foo"] }
   end
 
+  it "doesn't infer from new if generic" do
+    assert_error %(
+      class Foo(T)
+        def self.new
+          10
+        end
+      end
+
+      $x = Foo.new
+      $x
+      ),
+      "undefined global variable '$x'"
```

Comment:
@jhass I see. Yes, right now whenever the compiler can't infer the type of a global/class/instance variable it says "undefined variable" and then a long message explaining the rules the compiler uses to infer it. But I agree the message is misleading, I'll change it to "can't infer the type of ...", thanks.